lib/videoroom/room.ex

```diff
@@ -210,15 +185,34 @@ defmodule Videoroom.Room do
         state.peer_channels
         |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
 
-      Engine.remove_peer(state.rtc_engine, peer_id)
+      Membrane.Logger.info("Peer #{inspect(peer_id)} left RTC Engine")
+
+      Engine.remove_endpoint(state.rtc_engine, peer_id)
       {_elem, state} = pop_in(state, [:peer_channels, peer_id])
 
-      if state.peer_channels == %{}, do: Engine.terminate(state.rtc_engine)
+      if state.peer_channels == %{} do
+        Engine.terminate(state.rtc_engine)
+        {:stop, :normal, state}
+      else
+        {:noreply, state}
+      end
+    end
+  end
+
+  defp filter_codecs({%{encoding: "H264"}, fmtp}) do
+    import Bitwise
 
-      {:noreply, state}
+    case fmtp.profile_level_id >>> 16 do
+      0x42 -> (fmtp.profile_level_id &&& 0x00_4F_00) == 0x00_40_00
+      0x4D -> (fmtp.profile_level_id &&& 0x00_8F_00) == 0x00_80_00
+      0x58 -> (fmtp.profile_level_id &&& 0x00_CF_00) == 0x00_C0_00
```

Comment:
I'd would be nice to add a comment on that - what profiles are we selecting here?